src/libcore/fmt/mod.rs

```diff
@@ -529,6 +529,29 @@ pub trait Debug {
 #[stable(feature = "rust1", since = "1.0.0")]
 pub trait Display {
     /// Formats the value using the given formatter.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use std::fmt;
+    ///
+    /// enum Foo {
+    ///     Bar,
+    ///     Stack,
+    /// }
+    ///
+    /// impl fmt::Display for Foo {
+    ///     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+    ///         match *self {
+    ///             Foo::Bar => write!(f, "Foo::Bar"),
+    ///             Foo::Stack => write!(f, "Foo::Stack"),
```

Comment:
I think you're mixing up `Debug` and `Display`. `Display` is for **user-facing** output. Users shouldn't care about `enum` namespacing. We don't want to encourage Rust developers to implement `Display` if their types aren't mean to be displayed to end-users. 
How about this as the example:

```rust
struct Position {
    longitude: f32,
    latitude: 32,
}

impl fmt::Display for Position {
    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
        write!(f, "({}, {})", self.longitude, self.latitude)
    }
}
```

This seems a little more realistic than `Foo` and `Bar` and `Stack` ;)